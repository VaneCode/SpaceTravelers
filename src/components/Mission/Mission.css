table,
th,
td,
tr {
  border: 0.5px solid gray;
  border-collapse: collapse;
}

table {
  width: 95%;
  margin: 2rem;
}

.btnMember {
  background-color: gray;
  border: none;
  border-radius: 8px;
  padding: 0.1rem;
  color: white;
  margin: 0 1rem;
  display: flex;
  justify-content: center;
  width: 10rem;
}

.btnNotMember {
  background-color: dodgerblue;
  border: none;
  border-radius: 5px;
  padding: 0.1rem;
  color: white;
  margin: 0 1rem;
  display: flex;
  justify-content: center;
  width: 10rem;
}

.btnJoin {
  color: grey;
  border: 2px solid grey;
  border-radius: 5px;
  margin: 0.5rem 1rem;
  display: flex;
  justify-content: center;
  width: 7rem;
  font-size: medium;
  padding: 0.4rem;
}

.btnLeave {
  background-color: white;
  color: red;
  border: 2px solid red;
  margin: 0.5rem 1rem;
  display: flex;
  justify-content: center;
  width: 7rem;
  padding: 0.5rem 0;
  font-size: medium;
  border-radius: 5px;
}

.nameHead {
  font-weight: 900;
  font-size: 1.3rem;
  display: flex;
  flex-direction: column;
  border-bottom: none;
  border-left: none;
  border-right: none;
  border-top: none;
}

.headTable {
  text-align: left;
  font-weight: 900;
  font-size: 1.3rem;
  padding: 0.4rem;
}

.madMax {
  background-color: #f8f8f8;
  padding: 0.4rem;
}
