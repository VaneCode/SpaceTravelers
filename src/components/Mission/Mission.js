import { useSelector, useDispatch } from 'react-redux';
import React, { useEffect } from 'react';
import { getMissionData, joinMission } from '../../redux/missions/missions';
import './Mission.css';

const Mission = () => {
  const missions = useSelector((state) => state.missions.mission);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!missions.length) {
      dispatch(getMissionData());
    }
  });

  return (
    <div>
      <table>
        <thead>
          <tr>
            <th className="headTable">Mission</th>
            <th className="headTable">Description</th>
            <th className="headTable">Status</th>
          </tr>
        </thead>
        <tbody className="tableBody">
          {missions.map((mission, index) => (
            <tr key={mission.mission_id} className={(index + 1) % 2 === 0 && 'madMax'}>
              <td className="nameHead">{mission.mission_name}</td>
              <td>{mission.description}</td>
              <td>
                <button
                  type="button"
                  className={mission.join ? 'btnNotMember' : 'btnMember'}
                >
                  {mission.join ? 'ACTIVE MEMBER' : 'NOT A MEMBER'}
                </button>
              </td>
              <td>
                <button
                  type="button"
                  className={mission.join ? 'btnLeave' : 'btnJoin'}
                  onClick={() => dispatch(joinMission({ mission }))}
                >
                  {mission.join ? 'Leave Mission' : 'Join Mission'}
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Mission;
