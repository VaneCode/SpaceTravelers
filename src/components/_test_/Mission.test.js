import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import Mission from '../Mission/Mission';
import store from '../../redux/configureStore';

describe('mission test', () => {
  test('snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Mission />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
