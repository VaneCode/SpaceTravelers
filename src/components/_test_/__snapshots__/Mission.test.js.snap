// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mission test snapshot 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th
          className="headTable"
        >
          Mission
        </th>
        <th
          className="headTable"
        >
          Description
        </th>
        <th
          className="headTable"
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody
      className="tableBody"
    />
  </table>
</div>
`;
